src/snowflake/snowpark/functions.py

```diff
@@ -8700,3 +8704,135 @@ def make_interval(
+        func: A function to be applied to every row of the DataFrame.
+        output_types: A list of types for values generated by the ``func``
+        output_column_names: A list of names to be assigned to the resulting columns.
+        wrap_row: A flag to determine if the current row must be wrapped in an instance of the :class:`Row` class.
+        kwargs: Additional arguments to register the UDTF used to apply the Python function.
+
+    Example 1::
+
+        >>> from snowflake.snowpark.types import IntegerType
+        >>> from snowflake.snowpark.functions import map
+        >>> df = session.create_dataframe([[10, "a"]], schema=["col1", "col2"])
+        >>> new_df = map(df, lambda row: row[0] * row[0], output_types=[IntegerType()])
+        >>> new_df.show()
+        ---------
+        |"C_1"  |
+        ---------
+        |100    |
+        ---------
+        <BLANKLINE>
+
+    Example 2::
+
+        >>> new_df = map(df, lambda row: (row[1], row[0] * 3), output_types=[StringType(), IntegerType()])
+        >>> new_df.show()
+        -----------------
+        |"C_1"  |"C_2"  |
+        -----------------
+        |a      |30     |
+        -----------------
+        <BLANKLINE>
+
+    Example 3::
+
+        >>> new_df = map(df, lambda row: (row[1], row[0] * 3), output_types=[StringType(), IntegerType()], output_column_names=['col1', 'col2'])
+        >>> new_df.show()
+        -------------------
+        |"COL1"  |"COL2"  |
+        -------------------
+        |a       |30      |
+        -------------------
+        <BLANKLINE>
+
+    Example 4::
+        >>> df = session.create_dataframe([10, 20, 30])
+        >>> new_df = map(df, lambda x: x * x, output_types=[IntegerType()], wrap_row=False)
+        >>> new_df.sort(new_df[0]).show()
+        ---------
+        |"C_1"  |
+        ---------
+        |100    |
+        |400    |
+        |900    |
+        ---------
+        <BLANKLINE>
+
+    Note:
+        The result of the `func` function must be either a scalar value or a tuple containing the same number of elements
+        as specified in the `output_types` argument.
+    """
+
+    kwargs["packages"] = add_snowpark_package_to_sproc_packages(
+        dataframe._session, kwargs.get("packages")
+    )
+    if len(output_types) == 0:
+        raise ValueError("output_types cannot be empty.")
+    input_types = [field.datatype for field in dataframe.schema.fields]
+    num_fields = len(dataframe.schema.fields)
+
+    if output_column_names is None:
+        output_column_names = [f"c_{i+1}" for i in range(len(output_types))]
+    elif len(output_column_names) != len(output_types):
+        raise ValueError(
+            "'output_column_names' and 'output_types' must be of the same size."
+        )
+
+    output_schema = [
+        StructField(name, type_)
+        for name, type_ in zip(output_column_names, output_types)
+    ]
+
+    output_cols = [
+        col(f"${i + num_fields + 1}").alias(col_name)
```

Comment:
```suggestion
        col(f"${i + num_fields + 1}").alias(col_name)  # this is done to avoid collision with original table columns
```

Reply:
Fixed in the commit [fbeaf06](https://github.com/snowflakedb/snowpark-python/pull/2378/commits/fbeaf06cf9e062c0860e6737ad82faf9fcaf939a)